ambassador/ambassador/ir/irauth.py

```diff
@@ -121,10 +122,16 @@ def _load_auth(self, module: Resource):
         self["proto"] = module.get("proto", "http")
         self["timeout_ms"] = module.get("timeout_ms", 5000)
         self["connect_timeout_ms"] = module.get("connect_timeout_ms", 3000)
+        self["status_on_error"] = module.get("status_on_error", None)
+        self["failure_mode_allow"] = module.get("failure_mode_allow", None)
+        self["retry_policy"] = module.get("retry_policy", None)
         self.__to_header_list('allowed_headers', module)
         self.__to_header_list('allowed_request_headers', module)
         self.__to_header_list('allowed_authorization_headers', module)
 
+        if not self["retry_policy"] == None:
```

Comment:
I had to read this twice to make sure I knew its effect. :wink: The typical Ambassador idiom here is 

`if self.get('retry_policy'):`

which covers a variety of paranoid corner cases.